### ISSUE TITLE
Issues with pandas.SparseDataFrame

### ISSUE THREAD

**Issue by jorahn** [wesm/feather#221]
I just ran into some issues with feather and sparse pandas DataFrames (Python 3.5). `feather.write_dataframe()` doesn't raise exceptions but `feather.read_dataframe()` might drop values (both v0.2 and v0.3) or raise a ValueError (v0.2) / a FeatherError (v0.3):

```
>>> import pandas as pd
>>> import feather
>>> import numpy as np
>>> df = pd.DataFrame([[0,np.nan],[np.nan,1]]).to_sparse()
>>> df.info()
<class 'pandas.sparse.frame.SparseDataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 2 columns):
0    1 non-null float64
1    1 non-null float64
dtypes: float64(2)
memory usage: 96.0 bytes
>>> df
     0    1
0  0.0  NaN
1  NaN  1.0
>>> feather.write_dataframe(df, 'test.feather')
>>> df2 = feather.read_dataframe('test.feather')
>>> df2
     0    1
0  0.0  1.0
```
The NaN values were dropped and row 0 and 1 merged into one row.

```
>>> df = pd.DataFrame([[0,1],[1,1]]).to_sparse(fill_value=1)
>>> df
     0    1
0  0.0  1.0
1  1.0  1.0
>>> feather.write_dataframe(df, 'test.feather')
>>> df2 = feather.read_dataframe('test.feather')
```
at this point v0.2 raises a ValueError:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../site-packages/feather/api.py", line 61, in read_dataframe
    return pd.DataFrame(data, columns=names)
  File "/.../site-packages/pandas/core/frame.py", line 224, in __init__
    mgr = self._init_dict(data, index, columns, dtype=dtype)
  File "/.../site-packages/pandas/core/frame.py", line 318, in _init_dict
    extract_index(list(data.values()))
  File "/.../site-packages/pandas/core/frame.py", line 5279, in extract_index
    raise ValueError('arrays must all be same length')
ValueError: arrays must all be same length
```
while v0.3 raises a FeatherError instead:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../site-packages/feather/api.py", line 58, in read_dataframe
    reader = ext.FeatherReader(path)
  File "feather/ext.pyx", line 214, in feather.ext.FeatherReader.__cinit__ (feather/ext.cpp:3704)
  File "feather/ext.pyx", line 55, in feather.ext.check_status (feather/ext.cpp:1666)
feather.ext.FeatherError: Invalid: File is smaller than indicated metadata size
```

Are sparse structures generally incompatible with feather-format? If so, maybe there should be at least a warning on feather.write_dataframe(), when a SparseDataFrame is provided. Or is it possible/sensible to make this work?

BR,
Jonathan

**Comment by wesm** [comment]
Well, they definitely are not tested! I would welcome a PR with test cases and a conversion from sparse to dense prior to writing to Feather format. I don't expect native support for sparse structures is on the horizon (unless a motivated person does the implementation work)

**Comment by jorahn** [comment]
sure, please take a look

**Comment by jorahn** [comment]
@wesm anything missing in #223 ? let me know if anything else is required.

**Comment by wesm** [comment]
Just one small issue and I can merge

**Comment by jorahn** [comment]
done

**Comment by jorahn** [comment]
might take a look @wesm ?

**Comment by wesm** [comment]
fixed by #223
